source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -123,14 +123,14 @@ SystemTime Utility::getValidFrom(const X509& cert) {
   int days, seconds;
   int rc = ASN1_TIME_diff(&days, &seconds, &epochASN1_Time(), X509_get0_notBefore(&cert));
   ASSERT(rc == 1);
-  return std::chrono::system_clock::from_time_t(days * 24 * 60 * 60 + seconds);
+  return std::chrono::system_clock::from_time_t(static_cast<time_t>(days) * 24 * 60 * 60 + seconds);
```

Comment:
I'm a little confused here. Does the returned int go negative since internally it's 32-bit, and then we cast it back to positive 64-bit? Can you add a comment here and below?

Reply:
What's 32-bit?

The point of this change, as far as I understand is, is that previously, we had:
`32-bit int * 32-bit int * 32-bit int * 32-bit int + 32-bit int` multiplication, so the result was `32-bit int`.

With first argument being explicitly converted to `time_t` (64-bit), we have:
`64-bit time_t * 32-bit int * 32-bit int * 32-bit int + 32-bit int` multiplication, so the result is `64-bit time_t`.

Reply:
Right my point is the API is returning a 32-bit value, which has wrapped and gone negative, and we are casting it back to 64-bit unsigned? Right? I think this deserves a comment.

Reply:
You mean that `days` is wrapped and gone negative? I don't believe that's the case.

The result of multiplication overflows 32-bit and goes negative, and casting to 64-bit prevents that from happening.

Reply:
Ah I see, OK. Anyway, the length of this comment thread means we need a comment in the code. :)